src/unix/linux.c

```diff
@@ -2281,6 +2281,124 @@ uint64_t uv_get_available_memory(void) {
 }
 
 
+#define BUF_SIZE 1024
+
+static uv__cpu_constraint uv__get_cgroupv2_constrained_cpu(const char* cgroup) {
+  uv__cpu_constraint constraint = {0, 0, 0};
+  char path[256];
+  char buf[BUF_SIZE];
+  unsigned int weight;
+  int cgroup_size;
+  const char* cgroup_trimmed;
+
+  /* Trim ending \n by replacing it with a 0 */
+  cgroup_trimmed = cgroup + sizeof("0::/") - 1;      /* Skip the prefix "0::/" */
+  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first slash */
+
+  /* Construct the path to the cpu.max file */
+  snprintf(path, sizeof(path), "/sys/fs/cgroup/%.*s/cpu.max", cgroup_size,
+           cgroup_trimmed);
+
+  /* Read cpu.max */
+  if (uv__slurp(path, buf, sizeof(buf)) >= 0) {
```

Comment:
If you can't read cpu.max, there's no point in going on to read cpu.weight, right? Just bail out.

This function should probably return an error instead of a struct so the caller can properly distinguish between success and error. Returning a partially initialized success value and leaving it up to the caller to figure out what's what is not a good idea.

Reply:
OK, should I add an extra property in `uv__cpu_constraint` to flag it's an error ?